lms/static/scripts/frontend_apps/config.js

```diff
@@ -82,31 +90,37 @@ import { createContext } from 'preact';
  */
 
 /**
- * @typedef {'blackboard_missing_integration' |
- *  'canvas_invalid_scope' } OAuthErrorCode
- */
-
-/**
- * Configuration for the error dialog shown if authorizing access to an
- * OAuth API fails.
- *
- * @typedef OAuthErrorConfig
- * @prop {string|null} authUrl
- * @prop {string} errorDetails
- * @prop {OAuthErrorCode|null} errorCode
- * @prop {string[]} canvasScopes
+ *
+ * @typedef {'blackboard_missing_integration' | 'canvas_invalid_scope' | 'reused_consumer_key'} ConfigErrorCode
+ *
+ * Base interface for error information provided through this configuration object.
+ *
+ * @typedef ConfigErrorBase
+ * @prop {ConfigErrorCode} [errorCode]
+ * @prop {object|string} [errorDetails]
  */
 
 /**
- * @typedef {'reused_consumer_key'} ErrorCode
+ * Configuration for information describing general errors for use in
+ * `error-dialog` mode.
+ *
+ * @typedef ErrorDialogConfigBase
+ * @prop {Extract<'reused_consumer_key', ConfigErrorCode>} [errorCode]
```

Comment:
`Extract` seems to do the trick relatively nicely here.

Reply:
What was the intention of `Extract` here? Using `@prop {'reused_consumer_key'} [errorCode]` should I think have the same result.

Reply:
I was trying to establish that it was a subset of all known `ConfigErrorCode`s instead of just adding a new string prop to each variant. However, in looking at it with less exhausted face and eyes this morning, it may be more chattery than valuable and I may well do just what you've suggested here.